### ISSUE TITLE
softmax activation outputs value greater than 1

### ISSUE THREAD

**Issue by itayzach** [hls-fpga-machine-learning/hls4ml#115]
Hi,

First off, good job on the entire hls4ml package! I'm using it in a project of mine and it's great.

I believe there's an issue with the softmax activation. 
First row is the input, second row is the output.:

![bug](https://user-images.githubusercontent.com/13486006/47563129-69fee880-d929-11e8-96ae-64806d038881.PNG)

I expected the output to be less than 1 in all positions, but the second position which is 1 as expected.
Note that 1.33 isn't feasible for softmax by definition.

Reproduce:
- All variables are ap_fixed<32,8>
- The configuration struct is:
```C
#define M_in 4
typedef ap_fixed<32,8> result_t;
...
struct dec_softmax_config2 : nnet::activ_config {
    static const unsigned n_in = M_in;
    static const unsigned table_size = 1024;
    static const unsigned io_type = nnet::io_parallel;
};
```
- The call to softmax is
```C
result_t logits2[M_in];
#pragma HLS ARRAY_PARTITION variable=logits2 complete dim=0
result_t logits3[M_in];
#pragma HLS ARRAY_PARTITION variable=logits3 complete dim=0
nnet::softmax<result_t, result_t, dec_softmax_config2>(logits2, logits3);
```

Maybe I'm using it incorrectly?
Let me know if you need more info.

Thanks.

**Comment by itayzach** [comment]
I increased the table_size to 2048 and got the expected output:
![bug_fix](https://user-images.githubusercontent.com/13486006/47566317-0af2a100-d934-11e8-812c-4be6182721f9.PNG)

Thank you :+1:
